src/Raven.Studio/typescript/common/typeUtils.ts

```diff
@@ -65,3 +65,17 @@ export function compareSets<T extends string | number>(set1: T[], set2: T[]): bo
     }
     return true;
 }
+
+export function capitalize<Value extends string, Result extends Capitalize<Lowercase<Value>>>(value: Value): Result {
+    if (value == null) {
+        return "" as Result;
+    }
+
+    const stringValue = String(value);
+
+    if (!stringValue.length) {
+        return "" as Result;
+    }
+
+    return (stringValue.toLowerCase()[0].toUpperCase() + stringValue.slice(1).toLowerCase()) as Result;
```

Comment:
`toUpperCase` and `toLowerCase` returns string type so I need to cast here. Same for `""`.